File: TeeNew/Export_VML.pas
unit Export_VML;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series;

type
  TExportVMLForm = class(TBaseForm)
    Series1: TBarSeries;
    Series2: TBarSeries;
    Series3: TBarSeries;
    Series4: TBarSeries;
    Button1: TButton;
    SaveDialog1: TSaveDialog;
    Button2: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
    Procedure ShowSavedFile;
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeVMLCanvas, TeExport;

procedure TExportVMLForm.Button1Click(Sender: TObject);
begin
  if SaveDialog1.Execute then
  begin
    With TVMLExportFormat.Create do
    try
      Panel:=Chart1;
      SaveToFile(SaveDialog1.FileName);
      ShowSavedFile;
    finally
      Free;
    end;
  end;
end;

procedure TExportVMLForm.Button2Click(Sender: TObject);
begin
  {$IFNDEF D7}
  TeeExport(Self,Chart1)
  {$ENDIF}
end;

procedure TExportVMLForm.FormCreate(Sender: TObject);
begin
  inherited;
  Chart1.SeriesList.FillSampleValues(6);
end;

Procedure TExportVMLForm.ShowSavedFile;
begin
  TeeGoToURL(Handle,SaveDialog1.FileName);
end;

initialization
  RegisterClass(TExportVMLForm);
end.

File: TeeNew/Builder2009/Export_VML.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Export_VML.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeExport"
#pragma link "TeeVMLCanvas"
#pragma resource "*.dfm"
TExportVMLForm *ExportVMLForm;
//---------------------------------------------------------------------------
__fastcall TExportVMLForm::TExportVMLForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TExportVMLForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(6);
  Series2->FillSampleValues(6);
  Series3->FillSampleValues(6);
  Series4->FillSampleValues(6);
}
//---------------------------------------------------------------------------
void __fastcall TExportVMLForm::ShowSavedFile(void)
{
  TeeGotoURL(0,SaveDialog1->FileName);
}

void __fastcall TExportVMLForm::Button1Click(TObject *Sender)
{
  if (SaveDialog1->Execute())
  {
    TVMLExportFormat *Export=new TVMLExportFormat();
    Export->Panel=Chart1;
    Export->SaveToFile(SaveDialog1->FileName);
    ShowSavedFile();
    delete Export;
  }
}
//---------------------------------------------------------------------------

void __fastcall TExportVMLForm::Button2Click(TObject *Sender)
{
  TeeExport(this,Chart1);
}
//---------------------------------------------------------------------------

